src/esp/scene/Mp3dSemanticScene.cpp

```diff
@@ -94,29 +94,36 @@ bool SemanticScene::loadMp3dHouse(
 
   const bool hasWorldRotation = !rotation.isApprox(quatf::Identity());
 
-  auto getVec3f = [&](const std::vector<std::string>& tokens, int offset) {
+  auto getVec3f = [&](const std::vector<std::string>& tokens, int offset,
+                      bool applyRotation = true) {
     const float x = std::stof(tokens[offset]);
     const float y = std::stof(tokens[offset + 1]);
     const float z = std::stof(tokens[offset + 2]);
     vec3f p = vec3f(x, y, z);
-    if (hasWorldRotation) {
+    if (applyRotation && hasWorldRotation) {
       p = rotation * p;
     }
     return p;
   };
 
   auto getBBox = [&](const std::vector<std::string>& tokens, int offset) {
-    return box3f(getVec3f(tokens, offset), getVec3f(tokens, offset + 3));
+    return box3f(getVec3f(tokens, offset),
+                 getVec3f(tokens, offset + 3).array().abs().matrix());
```

Comment:
This seems unrelated to the rotation. Different bug?

Reply:
Oh yeah, just saw that we still hadn't fixed this -- after rotation, the sizes may be negative, which is weird -- so just did it.